pkg/controllers/cluster/import.go

```diff
@@ -139,6 +137,20 @@ func (i *importHandler) deleteOldAgent(cluster *fleet.Cluster, kc kubernetes.Int
 		}
 	}
 
+	if i.systemNamespace != config.DefaultNamespace {
+		if _, err := kc.CoreV1().Namespaces().Get(i.ctx, config.DefaultNamespace, metav1.GetOptions{}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				logrus.Infof("Deleted old agent for cluster %s/%s", cluster.Namespace, cluster.Name)
+				return nil
+			}
+			return err
+		}
+		if err := kc.AppsV1().Deployments(config.DefaultNamespace).Delete(i.ctx, config.AgentConfigName, metav1.DeleteOptions{}); err != nil {
```

Comment:
Same for here. If err != nil && errors.isNotFound(err) then return error. Otherwise this will fail if fleet-system exist but no fleet-agent deployment.

Reply:
You can test if this should enough, cover both case if namespace and deployment doesn't exist. If both doesn't exist then we should do nothing, which is the case in fresh install.
```
if err := kc.AppsV1().Deployments(config.DefaultNamespace).Delete(i.ctx, config.AgentConfigName, metav1.DeleteOptions{}); err != nil && errors.isNotFound(err) {
```

Reply:
I'll return nil in order to avoid the logging message if it's not found.